dot/peerset/peerset.go

```diff
@@ -356,7 +356,7 @@ func (ps *PeerSet) reportPeer(change ReputationChange, peers ...peer.ID) error {
 			// disconnect peer
 			err = ps.peerState.disconnect(i, pid)
 			if err != nil {
-				return fmt.Errorf("cannot disconnect peer %s at set %d: %w", pid, i, err)
+				return fmt.Errorf("cannot disconnect: %w", err)
```

Comment:
nit add the set context to the error wrapping in `fmt.Errorf("%w: for peer id %s", ErrPeerDoesNotExist, peerID)` in the disconnect method